### PR TITLE
tools: Add script to detect modules not called in any test plan

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by kravciak** [comment]
This will show qa-automation tests that are run from variable
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/products/sle/main.pm#L1054

**Comment by okurz** [comment]
ah, that's why they show up. Hm, so maybe a better approach is to just execute the main.pm and break early before actually executing tests but I guess as a first shot the script will help with some whitelisting

**Comment by kravciak** [comment]
Also you will have false-positives with dependencies, like:
tests/virt_autotest/host_upgrade_base.pm
 -> tests/virt_autotest/guest_installation_run.pm: use base "host_upgrade_base";

**Comment by okurz** [comment]
Well, but any "base" should be in "lib", I guess?

File: tools/detect_unused_modules [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh -e
+
+# checkout sle11 test plan to also check for modules used in there. sle11
+# should be the only other older testplan which might use test modules from
+# master
+git co origin/sle11_only_testplan -- main.pm
```
Comment (mimi1vx) [inline tools/detect_unused_modules]: ? please don't use your local aliases to standard commands

Comment (okurz) [inline tools/detect_unused_modules]: yes, makes sense. Ok, let's wait for https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/2096 and then we shouldn't need this anymore

Comment (okurz) [inline tools/detect_unused_modules]: fixed: sle11 is now a self-contained branch, i.e. I removed this special behaviour from the detect script

**Comment by okurz** [comment]
−3,351 lines of code, I love removing code :-)

Updated after sle11 was turned into a fully self-contained branch now. If you can not approve this PR I could remove some commits and we start small but please don't leave it around dangling :-)

**Comment by nilxam** [comment]
I'm sure your script does not proper aware of ```loadtest "x11/${de}_reconfigure_openqa.pm";```

**Comment by coolo** [comment]
and removing yast2_nis sounds a bit premature - IMO: remove sle11 only and do not add a check

**Comment by okurz** [comment]
> IMO: remove sle11 only and do not add a check

please see #2109

**Pull request closed without merging**